### ISSUE TITLE
motors spin up on save and reboot 

### ISSUE THREAD

**Issue by pkdevel** [cleanflight/cleanflight#1031]
whenever i click on 'save and reboot' or save via cli the motors spin up very shortly
im using 1.9.0 and oneshot. i see that there is only one initialization of oneshot now, maybe this is a regression.
if i can help providing more detailed information, please give me a hint

**Comment by bssstudio** [comment]
Maybe it will be useful to know what kind of ESCs (ESC model and firmware) are you using.

I had similar issues with oneshot. The motors would twitch randomly and I ended up lowering the min command to about 900 or so.

**Comment by hydra** [comment]
@pkdevel loads of work was put into this over the last couple of releases so im suprised youre having issues.  The double init was fixed too.

Need config dump and EXACT details of your escs and their settings before anything can be done.

**Comment by borisbstyle** [comment]
Yeah very curious to how you can still have this issue

**Comment by pkdevel** [comment]
hey guys,

sorry for the late response.
i'm using the ultra escs 20a with default settings, except that oneshot and active fw is turned on.
cant provide a dump right now, maybe tonight.

i have to say that i merged boris' "pt1 element" pr into the 1.9.0 version.
but i'm using the exact same version on multiple systems (naze and cc3d) without any issues,
except for two setups: one naze with those ultras and a default emax 250 setup with a cc3d.

**Comment by pkdevel** [comment]
just found a dump on this machine, i think was the backup at this time, before i change my naze full to a naze acro:


# dump

# version
# Cleanflight/NAZE 1.9.0 Jun 15 2015 / 16:48:08 (98f7549)
# dump master

# mixer
mixer QUADX


# feature
feature -RX_PPM
feature -VBAT
feature -INFLIGHT_ACC_CAL
feature -RX_SERIAL
feature -MOTOR_STOP
feature -SERVO_TILT
feature -SOFTSERIAL
feature -GPS
feature -FAILSAFE
feature -SONAR
feature -TELEMETRY
feature -CURRENT_METER
feature -3D
feature -RX_PARALLEL_PWM
feature -RX_MSP
feature -RSSI_ADC
feature -LED_STRIP
feature -DISPLAY
feature -ONESHOT125
feature -BLACKBOX
feature VBAT
feature RX_SERIAL
feature MOTOR_STOP
feature FAILSAFE
feature RSSI_ADC
feature LED_STRIP
feature ONESHOT125


# map
map TAER1234


# serial
serial 0 1 115200 57600 0 115200
serial 1 64 115200 57600 0 115200


# led
led 0 3,15:S:AC:2
led 1 4,15:S:C:2
led 2 5,15:S:C:2
led 3 6,15:S:C:2
led 4 7,15:S:C:2
led 5 8,15:S:C:2
led 6 9,15:S:C:2
led 7 10,15:S:C:2
led 8 11,15:S:C:2
led 9 12,15:S:AC:2
led 10 0,0:::0
led 11 0,0:::0
led 12 0,0:::0
led 13 0,0:::0
led 14 0,0:::0
led 15 0,0:::0
led 16 0,0:::0
led 17 0,0:::0
led 18 0,0:::0
led 19 0,0:::0
led 20 0,0:::0
led 21 0,0:::0
led 22 0,0:::0
led 23 0,0:::0
led 24 0,0:::0
led 25 0,0:::0
led 26 0,0:::0
led 27 0,0:::0
led 28 0,0:::0
led 29 0,0:::0
led 30 0,0:::0
led 31 0,0:::0


# color
color 0 0,0,0
color 1 0,255,255
color 2 0,0,255
color 3 30,0,255
color 4 60,0,255
color 5 90,0,255
color 6 120,0,255
color 7 150,0,255
color 8 180,0,255
color 9 210,0,255
color 10 240,0,255
color 11 270,0,255
color 12 300,0,255
color 13 330,0,255
color 14 0,0,0
color 15 0,0,0

set looptime = 1400
set emf_avoidance = 0
set mid_rc = 1500
set min_check = 1050
set max_check = 1950
set rssi_channel = 0
set rssi_scale = 30
set rssi_ppm_invert = 0
set input_filtering_mode = 0
set min_throttle = 1100
set max_throttle = 2000
set min_command = 1000
set servo_center_pulse = 1500
set 3d_deadband_low = 1406
set 3d_deadband_high = 1514
set 3d_neutral = 1460
set 3d_deadband_throttle = 50
set motor_pwm_rate = 400
set servo_pwm_rate = 50
set retarded_arm = 0
set disarm_kill_switch = 0
set auto_disarm_delay = 5
set small_angle = 25
set flaps_speed = 0
set fixedwing_althold_dir = 1
set reboot_character = 82
set gps_provider = 0
set gps_sbas_mode = 0
set gps_auto_config = 1
set gps_auto_baud = 0
set serialrx_provider = 2
set spektrum_sat_bind = 0
set telemetry_switch = 0
set telemetry_inversion = 0
set frsky_default_lattitude =  0.000
set frsky_default_longitude =  0.000
set frsky_coordinates_format = 0
set frsky_unit = 0
set frsky_vfas_precision = 0
set hott_alarm_sound_interval = 5
set battery_capacity = 0
set vbat_scale = 111
set vbat_max_cell_voltage = 43
set vbat_min_cell_voltage = 32
set vbat_warning_cell_voltage = 35
set current_meter_scale = 400
set current_meter_offset = 0
set multiwii_current_meter_output = 0
set current_meter_type = 1
set align_gyro = 0
set align_acc = 0
set align_mag = 0
set align_board_roll = 0
set align_board_pitch = 0
set align_board_yaw = 0
set max_angle_inclination = 500
set gyro_lpf = 256
set moron_threshold = 32
set gyro_cmpf_factor = 600
set gyro_cmpfm_factor = 250
set yaw_control_direction = 1
set pid_at_min_throttle = 1
set yaw_direction = 1
set yaw_jump_prevention_limit = 200
set tri_unarmed_servo = 1
set servo_lowpass_freq = 400
set servo_lowpass_enable = 0
set failsafe_delay = 10
set failsafe_off_delay = 200
set failsafe_throttle = 1200
set rx_min_usec = 985
set rx_max_usec = 2115
set acc_hardware = 0
set mag_hardware = 0
set blackbox_rate_num = 1
set blackbox_rate_denom = 1
set blackbox_device = 1

# dump profile

# profile
profile 0

# aux
aux 0 0 2 1400 2100
aux 1 1 0 900 1300
aux 2 2 0 1300 1700
aux 3 12 1 1600 2100
aux 4 14 13 1300 2100
aux 5 18 5 1300 2100
aux 6 20 5 1700 2100
aux 7 0 0 900 900
aux 8 0 0 900 900
aux 9 0 0 900 900
aux 10 0 0 900 900
aux 11 0 0 900 900
aux 12 0 0 900 900
aux 13 0 0 900 900
aux 14 0 0 900 900
aux 15 0 0 900 900
aux 16 0 0 900 900
aux 17 0 0 900 900
aux 18 0 0 900 900
aux 19 0 0 900 900

# adjrange
adjrange 0 0 7 900 1300 6 8
adjrange 1 0 7 1300 1700 7 8
adjrange 2 0 7 1700 2100 8 8
adjrange 3 0 0 900 900 0 0
adjrange 4 0 0 900 900 0 0
adjrange 5 0 0 900 900 0 0
adjrange 6 0 0 900 900 0 0
adjrange 7 0 0 900 900 0 0
adjrange 8 0 0 900 900 0 0
adjrange 9 0 0 900 900 0 0
adjrange 10 0 0 900 900 0 0
adjrange 11 0 0 900 900 0 0

# servo
servo 0 1020 2000 1500 30 -1
servo 1 1020 2000 1500 30 -1
servo 2 1020 2000 1500 100 -1
servo 3 1020 2000 1500 100 -1
servo 4 1020 2000 1500 100 -1
servo 5 1020 2000 1500 100 -1
servo 6 1020 2000 1500 100 -1
servo 7 1020 2000 1500 100 -1
servo 8 1020 2000 1500 100 -1
servo 9 1020 2000 1500 100 -1

set gps_pos_p = 15
set gps_pos_i = 0
set gps_pos_d = 0
set gps_posr_p = 34
set gps_posr_i = 14
set gps_posr_d = 53
set gps_nav_p = 25
set gps_nav_i = 33
set gps_nav_d = 83
set gps_wp_radius = 200
set nav_controls_heading = 1
set nav_speed_min = 100
set nav_speed_max = 300
set nav_slew_rate = 30
set alt_hold_deadband = 40
set alt_hold_fast_change = 1
set deadband = 5
set yaw_deadband = 10
set throttle_correction_value = 0
set throttle_correction_angle = 800
set default_rate_profile = 0
set gimbal_flags = 1
set acc_lpf_factor = 4
set accxy_deadband = 40
set accz_deadband = 40
set accz_lpf_cutoff =  5.000
set acc_unarmedcal = 1
set acc_trim_pitch = 0
set acc_trim_roll = 0
set baro_tab_size = 21
set baro_noise_lpf =  0.600
set baro_cf_vel =  0.985
set baro_cf_alt =  0.965
set mag_declination = 0
set pid_controller = 1
set p_pitch = 46
set i_pitch = 35
set d_pitch = 50
set p_roll = 39
set i_roll = 35
set d_roll = 65
set p_yaw = 97
set i_yaw = 35
set d_yaw = 10
set p_pitchf =  2.200
set i_pitchf =  0.400
set d_pitchf =  0.030
set p_rollf =  2.000
set i_rollf =  0.350
set d_rollf =  0.025
set p_yawf =  3.500
set i_yawf =  0.200
set d_yawf =  0.015
set level_horizon =  3.000
set level_angle =  5.000
set sensitivity_horizon = 75
set p_alt = 50
set i_alt = 0
set d_alt = 0
set p_level = 20
set i_level = 15
set d_level = 100
set p_vel = 120
set i_vel = 45
set d_vel = 1
set yaw_p_limit = 500
set main_cut_hz = 17

# dump rates

# rateprofile
rateprofile 0

set rc_rate = 130
set rc_expo = 65
set rc_yaw_expo = 35
set thr_mid = 40
set thr_expo = 35
set roll_rate = 35
set pitch_rate = 35
set yaw_rate = 30
set tpa_rate = 50
set tpa_breakpoint = 1350

#

**Comment by borisbstyle** [comment]
that version is quite old. You need to try the latest master.

**Comment by pkdevel** [comment]
@borisbstyle just saw your pr #1028. i guess you're talking about this one.
gonna merge this tonight and give it a try.

will report asap.

cheers

**Comment by hydra** [comment]
Any update?

**Comment by pkdevel** [comment]
hey guys,

so i just tried out the plain 1.9.0 tag version with only #1028 merged into.
i still have the same issue:
https://youtu.be/TogyE-QSzQ0
https://youtu.be/fQSzll4NiLg

funny thing is, i tried with my dump and also with default settings. as soon as i enable oneshot and reboot a second time, the motors spin up.

i guess its sensless to post another cli dump, since there is only oneshot enabled.

here are my ultraesc settings:
<img width="779" alt="screen shot 2015-07-08 at 18 15 21" src="https://cloud.githubusercontent.com/assets/2041818/8577428/e258d88c-25a8-11e5-9abd-6d5bed01590f.png">

can i provide anything else to help?
any ideas how i could debug more?

cheers

**Comment by borisbstyle** [comment]
Is this the latest master or the official 1.9?
I cant remember.......but I though the final fix was in 1.9.
So its just ultra esc having this issue?
Op 8 jul. 2015 19:40 schreef "Philipp Kolb" <notifications@github.com>:

**Comment by pkdevel** [comment]
i'm using the offical/final 1.9.0 tag version.
i don't think this is an issue regarding those ultras … a friend using some cheap blhelis 12a has the same issue

**Comment by borisbstyle** [comment]
than it sounds like you are not using the version with the fix. There are
many people with blheli including me which dont have this issue anymore.
Please try the latest master or this one
https://www.dropbox.com/s/9690qnccqguhjre/cleanflight_NAZE.hex?dl=0

i'm using the offical/final 1.9.0 tag version.
i don't think this is an issue regarding those ultras … a friend using some
cheap blhelis 12a has the same issue

—
Reply to this email directly or view it on GitHub
<https://github.com/cleanflight/cleanflight/issues/1031#issuecomment-119680050>
.

**Comment by hydra** [comment]
no response, no further reports. closing.  please re-open a new issue and reference this if there is still a problem.

**Comment by paalmiso** [comment]
I think I've just experienced this with my IMRC Vortex. I'm on Cleanflight 1.9.1 downloaded from here http://www.immersionrc.com/fpv-products/vortex-racing-quad/#specifications My Vortex Firmware version is v1.0.0.15

How can I help you debug this?

I was editing configuration - checked checkbox to prevent motors spinning after arming. Then I've hit save and reboot, and my motors went full throttle for few seconds shredding skin on my little finger. I've ended on ER, fortunately nothing serious.
